packages/orm/src/relations/PolymorphicReference.ts

```diff
@@ -72,7 +72,10 @@ export class PolymorphicReferenceImpl<T extends Entity, U extends Entity, N exte
     private fieldName: keyof T & string,
   ) {
     super(entity);
-    this.field = getMetadata(entity).fields[this.fieldName] as PolymorphicField;
+    // If this polymorphic field is a subclass of another entity
+    const meta = getMetadata(entity);
+    const baseMeta = meta.baseTypes[0];
+    this.field = (baseMeta ?? meta).fields[this.fieldName] as PolymorphicField;
```

Comment:
Suggestion: It seems like `baseMeta ?? meta` will _always_ look at the base, when the field could still be on the subclass?

Fwiw I think changing the original `.fields[this.fieldName]` to just `.allFields[this.fieldName]` will probably work? `allFields` should already be setup have both base+sub fields merged together.